.cloudbuild/library_generation/library_generation.Dockerfile

```diff
@@ -47,7 +63,13 @@ RUN source /src/utils/utilities.sh \
 ENV DOCKER_GRPC_LOCATION="/grpc/protoc-gen-grpc-java-${GRPC_VERSION}-${OS_ARCHITECTURE}.exe"
 ENV DOCKER_GRPC_VERSION="${GRPC_VERSION}"
 
-# use python 3.11 (the base image has several python versions; here we define the default one)
+
+# we transfer gapic-generator-java from the previous stage.
+ENV DOCKER_GAPIC_GENERATOR_LOCATION="/gapic-generator-java/gapic-generator-java.jar"
```

Comment:
Do we need this location to be an env variable? Can this be inlined?

Reply:
We could set the jar into a well-known location. Developers would need to send their jar in there beforehand. 

I think this could solve the need for an env var and copying the jar?

In favor of the env var, I think it helps us separate the logic of the Dockerfile when it decides where to put the jar vs our scripts finding out where the jar is. But I'm not sure if this outweights the need to depend on an env var?

Copying the jar is more about keeping the changes minimal, without involving how the jar wrapper behaves. However, I think the reason we copy the jar into the output folder is because of the way we call the wrapper. 

https://github.com/googleapis/sdk-platform-java/blob/58da069a99a7240cbec18b7dbdb010fb74c4d551/library_generation/generate_library.sh#L209

We can configure the wrapper to expect the jar to be in the well-known location instead.

Reply:
We set the jar into a well-known location. No env vars used